woocommerce/woocommerce
# Linked product list

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #42923



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure to enable `product-linked` feature flag under `/wp-admin/tools.php?page=woocommerce-admin-test-helper`. WCA Test Helper plugin is required.
3. Go to `Products` -> `Add new` from the left side menu.
4. Under the `Linked products`, from the `Upsells` and `Cross-sells` now it's possible to search for a product. 
<img width="766" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/d54382ae-28d3-4a3b-a285-a4f6c121a18b">

5. Once select the product from the search select menu it will be shown in a table right below the select control.
6. From the table when clicking the name of the product, a redirection to the edit product page should happen in a new tab.
7. Form the table when clicking the arrow icon, a redirection to the product storefront page should happen in a new tab.
8. From the table when clicking the x icon, that product should be removed from the list.
9. All changes made should persist after saving the product and performing a hard refresh. 
<img width="722" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/bf70d53c-c912-43c4-8082-c4d240562dfa">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
